### PR TITLE
Add VIP Night module

### PR DESCRIPTION
**Pull request by pandathebeasty**

### CODE REVIEW

File: VIPCore/modules/VIP_NightVip/VIP_NightVip.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    {
+        if (_api == null) return;
+
+        var currentTime = DateTime.Parse(DateTime.Now.ToString("HH:mm:ss"));
+        var startTime = DateTime.Parse(Config.PluginStartTime);
+        var endTime = DateTime.Parse(Config.PluginEndTime);
+
+        if ((currentTime >= startTime || currentTime < endTime) && !_api.IsClientVip(player))
+        {
+            _api.GiveClientVip(player, Config.VIPGroup, -1);
```
Comment (partiusfabaa) [inline VIPCore/modules/VIP_NightVip/VIP_NightVip.cs]: VIPCore API has a function that gives a VIP for 1 session:
```cs
public void GiveClientTemporaryVip(CCSPlayerController player, string group, int time)
```

File: VIPCore/modules/VIP_NightVip/VIP_NightVip.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    private VIP_NightVipConfig LoadConfig()
+    {
+        var configPath = Path.Combine(_api.ModulesConfigDirectory, ConfigFileName);
+
+        if (!File.Exists(configPath)) return CreateConfig(configPath);
+
+        return JsonSerializer.Deserialize<VIP_NightVipConfig>(File.ReadAllText(configPath))!;
+    }
+
```
Comment (partiusfabaa) [inline VIPCore/modules/VIP_NightVip/VIP_NightVip.cs]: There is also a function in the VIPCore API for creating configs:

```cs
public T LoadConfig<T>(string name) // write name without .json
```

You just need to make the value in your class default, and write in the code
```cs
Config = _api.LoadConfig<VIP_NightVipConfig>("VIP_NightVip");
```

File: VIPCore/modules/VIP_NightVip/VIP_NightVip.cs (hunk 1 above)
Comment (pandathebeasty) [reply VIPCore/modules/VIP_NightVip/VIP_NightVip.cs]: So, how I can give for one round for example?

File: VIPCore/modules/VIP_NightVip/VIP_NightVip.cs (hunk 2 above)
Comment (pandathebeasty) [reply VIPCore/modules/VIP_NightVip/VIP_NightVip.cs]: Let me know how I can use this

Comment (pandathebeasty) [reply VIPCore/modules/VIP_NightVip/VIP_NightVip.cs]: Solved

File: VIPCore/modules/VIP_NightVip/VIP_NightVip.cs (hunk 1 above)
Comment (partiusfabaa) [reply VIPCore/modules/VIP_NightVip/VIP_NightVip.cs:113]: You can give for example for the time from which the person logged in, and until the end of `PluginEndTime`, If the person leaves, the vip will disappear

Comment (pandathebeasty) [reply VIPCore/modules/VIP_NightVip/VIP_NightVip.cs]: Check now

**Pull request merged**